x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_upgrade_modal/index.tsx

```diff
@@ -224,6 +243,29 @@ export const AgentUpgradeAgentModal: React.FunctionComponent<AgentUpgradeAgentMo
     }
   }, [latestAgentVersion]);
 
+  const warningMessage = useMemo(() => {
+    if (
+      isSingleAgent &&
+      selectedVersion[0]?.value &&
+      !isAgentUpgradeableToVersion(agents[0], selectedVersion[0].value)
```

Comment:
should we check here whether `agents` is string (query) or array?

Reply:
It's already checked in https://github.com/elastic/kibana/blob/24808a9b2c62bfc66c64f1f777c5363030b01178/x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_upgrade_modal/index.tsx#L96. This part didn't change in my commit.